src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/record-custom-events.mdx

```diff
@@ -0,0 +1,880 @@
+
+<Callout variant="important">
+  As of New Relic [Android agent version 5.12.0](/docs/release-notes/mobile-release-notes/android-release-notes/android-5120), the `recordEvent` method is deprecated and replaced with `recordCustomEvent`. The `recordEvent` method will continue to work for an unspecified period of time, but if your app contains `recordEvent` methods, New Relic recommends you replace them.
+
+  Updating these methods will affect any NRQL queries and <InlinePopover type="dashboards" /> that use the deprecated event types. Be sure to adjust your NRQL queries and dashboards as needed.
+</Callout>
+
+For more on other Android APIs, see [Send custom attributes and events](/docs/mobile-monitoring/new-relic-mobile-android/install-configure/guide-custom-instrumentation-api#send-insights).
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width="30%">
+        Parameter
+      </th>
+
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$eventType`
+      </td>
+
+      <td>
+        `string`
+      </td> 
+
+      <td>
+        Required. The type of event. **Do not** use `$eventType` to name your custom events. Instead, use a custom attribute or the optional `name`.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$eventName`
+      </td>
+
+      <td>
+        `dictionary`
+      </td>
+      <td>
+        Optional. A map that includes a list of attributes that further designate subcategories to the `$eventType`. You can create attributes for any event descriptors you think will be useful. To name your custom events, create a `name` attribute or use the `eventName` parameter.
+        
+        Note: Not all object types are supported. See [setAttribute](/docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/set-attribute/#parameters) for details on supported types.
+
+        <Callout variant="important">
+          When setting the key for your custom attributes, be aware that there are [default attributes that cannot be overridden](/docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/set-attribute#description).
+        </Callout>
+      </td>
+    </tr>
+
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if the event is recorded successfully, or `false` if not.
+
+## Examples [#examples]
+
+Here's an example of a basic custom event:
+
+### Java [#java] 
+
+```java
+Map attributes = new HashMap();
+attributes.put("attributeName1", "value1");
+attributes.put("attributeName1", 2);
+boolean eventRecorded = NewRelic.recordCustomEvent("eventType", attributes);
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
val attributes = mutableMapOf<String,Any>("attributeName1" to "value1", "attributeName1" to 2)
        val eventRecorded = NewRelic.recordCustomEvent("eventType", attributes)

Reply:
```suggestion
```kotlin

val attributes = mutableMapOf<String,Any>("attributeName1" to "value1", "attributeName1" to 2)
val eventRecorded = NewRelic.recordCustomEvent("eventType", attributes)

```
```